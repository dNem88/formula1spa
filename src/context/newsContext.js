import React from 'react';

const newsContext = React.createContext({articles: []})

export default newsContext;