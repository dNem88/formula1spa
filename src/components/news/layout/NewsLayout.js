import React, {useContext} from 'react'
import styles from './newsLayout.module.css'
import newsContext from '../../../context/newsContext'
import {Outlet} from 'react-router-dom'
import News from '../../common/news/News'


function NewsLayout() {
    const news = useContext(newsContext)
    console.log(news, 'NEWSLAYOUT')
    
    return (
        <section className={styles.wrapper}>
            <Outlet/>
            <div className={styles['news-container']}>
                <section onClick={() => {
                    window.scrollTo(0,0)
                }} className={styles['news-slider']}>
                    {news.articles ? news.articles.map(x => {
                        return <News key={x._id} id={x._id} imageUrl={x.urlToImage} title={x.title} hoverChange={true}/>
                    }) : null}
                </section>
            </div>
            
        </section>
    )
}

export default NewsLayout