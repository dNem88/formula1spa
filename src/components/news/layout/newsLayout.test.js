import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import NewsLayout from './NewsLayout'
import newsContext from '../../../context/newsContext'

jest.mock('../../common/video/Video', () => () => 'VideoMock')
jest.mock('../../common/videoinfo/VideoInfo', () => () => 'VideoInfoMock')

describe('Testing NewsLayout', () => {
    
    it('Render Properly', () => {
        const snapshot = renderer.create(
        <MemoryRouter>
            <newsContext.Provider value={{articles: [{
                _id: 'test-id',
                imageUrl: 'test-url',
                title: 'test-title',
                hoverChange: true
            }]}}>
                <NewsLayout />
            </newsContext.Provider>
        </MemoryRouter>
    ).toJSON()
        expect(snapshot).toMatchSnapshot()
        expect(snapshot.type).toBe('section')
        expect(snapshot.props.className).toBe('wrapper')
        expect(snapshot.children[0].type).toBe('div')
        expect(snapshot.children[0].props.className).toBe('news-container')
        let section = snapshot.children[0].children[0]
        expect(section.type).toBe('section')
        expect(section.props.className).toBe('news-slider')
        expect(section.children.length).toBe(1)
    })
})