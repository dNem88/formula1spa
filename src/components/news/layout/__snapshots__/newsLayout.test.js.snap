// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NewsLayout Render Properly 1`] = `
<section
  className="wrapper"
>
  <div
    className="news-container"
  >
    <section
      className="news-slider"
      onClick={[Function]}
    >
      <div
        className="main-container-vertical"
        id="test-id"
        onClick={[Function]}
      >
        <div
          className="image-container"
        >
          <img
            alt="image-from-newsapi"
            id="news-image"
            loading="lazy"
          />
        </div>
        <div
          className="dark-container"
        >
          <p
            className="news-par"
          >
            NEWS
          </p>
          <p
            className="content-par"
          >
            test-title
          </p>
        </div>
      </div>
    </section>
  </div>
</section>
`;
